tekton/pipeline_run_test.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2022 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tekton
+
+import (
+	"context"
+	"reflect"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/redhat-appstudio/release-service/api/v1alpha1"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ExtraParams struct {
+	Name  string
+	Value tektonv1beta1.ArrayOrString
+}
+
+type ReleasePipelineParams struct {
+	Prefix    string
+	Namespace string
+	Extra     ExtraParams
+}
+
+var _ = Describe("PipelineRun", func() {
+
+	var params *ReleasePipelineParams
+	var releasePipelineRun *ReleasePipelineRun
+
+	var release *v1alpha1.Release
+	var releaseParams *v1alpha1.ReleaseSpec
+
+	BeforeEach(func() {
+
+		params = &ReleasePipelineParams{
+			Prefix:    "mypipeline",
+			Namespace: "default",
+			// @todo: use a valid Pipeline params
+			Extra: ExtraParams{
+				Name:  "ExtraParam",
+				Value: tektonv1beta1.ArrayOrString{Type: tektonv1beta1.ParamTypeString, StringVal: "ExtraValue"},
+			},
+		}
+		releaseParams = &v1alpha1.ReleaseSpec{
+			ApplicationSnapshot: "mysnapshot",
+			ReleaseLink:         "myreleaselink",
+		}
+
+		release = &v1alpha1.Release{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "appstudio.redhat.com/v1alpha1",
+				Kind:       "Release",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "myrelease-",
+				Namespace:    params.Namespace,
+				UID:          "8ba3294a-2e1b-4a13-9b3a-faf3cfe3d31b", // This is an arbitrary random UUID
+			},
+			Spec: *releaseParams,
+		}
+
+		ctx := context.Background()
+
+		// The code below sets the ownership for the Release Object
+		kind := reflect.TypeOf(v1alpha1.Release{}).Name()
+		gvk := v1alpha1.GroupVersion.WithKind(kind)
+		controllerRef := metav1.NewControllerRef(release, gvk)
+
+		// creating a release
+		Expect(k8sClient.Create(ctx, release)).Should(Succeed())
+		release.SetOwnerReferences([]metav1.OwnerReference{*controllerRef})
+
+		// need to set the Kind as it loses it for reasons...
```

Comment:
I tested this and you're right. Do you know the reasons? If so, can we write them? If not, totally fine :slightly_smiling_face:  I personally have no clue

Reply:
The only "kind of" explanation I got researching it was: "EnvTest behaves differently from an actual cluster". 

https://github.com/kubernetes-sigs/controller-runtime/issues/1870

But I got some others (maybe) more _elegant_ ways to fix it (maybe as not as clear as this one). I will give them a try.

Reply:
Up to you. I am fine with you just putting a link to that issue in the comments. If you want to spend more time on it and do the elegant fix, that works too

Reply:
I will keep it like this for the moment. Adding it manually is simple and clear to understand. If required in the future I can change it.